src/test/java/seedu/address/logic/parser/EditLogCommandParserTest.java

```diff
@@ -0,0 +1,226 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_LOG_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NEW_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NEW_TITLE;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_LOG;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_LOG;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.EditLogCommand;
+import seedu.address.model.person.FriendName;
+import seedu.address.model.person.LogName;
+
+
+public class EditLogCommandParserTest {
+    private static final EditLogCommandParser parser = new EditLogCommandParser();
+    private static final String MESSAGE_INVALID_FORMAT = EditLogCommandParser.MESSAGE_INVALID_FORMAT;
+    private static final String MESSAGE_INVALID_TITLE = LogName.MESSAGE_CONSTRAINTS;
+    private static final String MESSAGE_INVALID_INDEX = ParserUtil.MESSAGE_INVALID_INDEX;
+
+
+    @Test
+    public void parse_invalidFormat_failure() {
+
+        Index personIndex = INDEX_FIRST_PERSON;
+        Index logIndex = INDEX_FIRST_LOG;
+        String args;
+
+        // missing person index/name -> fail
+        args = " " + PREFIX_LOG_INDEX + logIndex.getOneBased()
+                + " " + PREFIX_NEW_TITLE + VALID_LOG_TITLE;
+        assertParseFailure(parser, args, MESSAGE_INVALID_FORMAT);
+
+        // missing log index -> fail
+        args = personIndex.getOneBased()
+                + " " + PREFIX_NEW_TITLE + VALID_LOG_TITLE;
+        assertParseFailure(parser, args, MESSAGE_INVALID_FORMAT);
+
+        // missing both new title and new description
+        args = personIndex.getOneBased() + " " + PREFIX_LOG_INDEX + logIndex.getOneBased();
+        assertParseFailure(parser, args, MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidFieldValues_failure() {
+
+        Index personIndex = INDEX_FIRST_PERSON;
+        Index logIndex = INDEX_FIRST_LOG;
+        String args;
+
+        // invalid new title
+        args = personIndex.getOneBased() + " " + PREFIX_LOG_INDEX + logIndex.getOneBased() + " "
+                + PREFIX_NEW_TITLE + "";
+        assertParseFailure(parser, args, MESSAGE_INVALID_TITLE);
+
+        // invalid person index
+        args = -1 + " " + PREFIX_LOG_INDEX + logIndex.getOneBased()
+                + " " + PREFIX_NEW_TITLE + VALID_LOG_TITLE;
+        assertParseFailure(parser, args, MESSAGE_INVALID_INDEX);
+
+        // invalid log index
+        args = personIndex.getOneBased() + " " + PREFIX_LOG_INDEX + "-1"
+                + " " + PREFIX_NEW_TITLE + VALID_LOG_TITLE;
+        assertParseFailure(parser, args, MESSAGE_INVALID_INDEX);
```

Comment:
just a nitpick, maybe we can a test case for invalid description? Like when the user types in /d but without entering any description?

Reply:
Yep, done!

Reply:
ok great! 👍